pkgs/data/themes/tokyo-night-gtk/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, gnome-themes-extra
+, gtk-engine-murrine
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "tokyo-night-gtk";
+  version = "2023.01.17";
+
+  src = fetchFromGitHub {
+    owner = "Fausto-Korpsvart";
+    repo = "Tokyo-Night-GTK-Theme";
+    rev = "f7ae3421ac0d415ca57fb6224e093e12b8a980bb";
+    sha256 = "sha256-90V55pRfgiaP1huhD+3456ziJ2EU24iNQHt5Ro+g+M0=";
+  };
+
+  propagatedUserEnvPkgs = [
+    gtk-engine-murrine
+  ];
+
+  buildInputs = [
+    gnome-themes-extra
```

Comment:
Why add this to buildInputs?

Reply:
> Why add this to buildInputs?

It's part of the installing theme description on the [GH Page](https://github.com/Fausto-Korpsvart/Tokyo-Night-GTK-Theme#installing-themes)

Reply:
But as you have set `dontBuild = true;`, adding it should have no visible effect.

Reply:
> But as you have set `dontBuild = true;`, adding it should have no visible effect.

Makes sense, i guess `dontBuild = true` should be removed then.

Reply:
As this package is not actually built (just copying the files to output), I think removing buildInputs altogether is the right solution. Please try and test.

Reply:
> As this package is not actually built (just copying the files to output), I think removing buildInputs altogether is the right solution. Please try and test.

my reasoning behind it is, that the already [existing catpuccin-gtk-theme](https://github.com/NixOS/nixpkgs/blob/nixos-22.11/pkgs/data/themes/catppuccin-gtk/default.nix) should also be adjusted, unless it's different?

Reply:
Trying building it with the patch below applied:
```
--- a/pkgs/data/themes/catppuccin-gtk/default.nix
+++ b/pkgs/data/themes/catppuccin-gtk/default.nix
@@ -40,7 +40,6 @@ stdenvNoCC.mkDerivation rec {
   nativeBuildInputs = [ gtk3 sassc ];

   buildInputs = [
-    gnome-themes-extra
     (python3.withPackages (ps: [ ps.catppuccin ]))
   ];
```
And there's no difference in the output according to `diff -r`
Yes, that package should also be adjusted, maybe together with many other theme packages.

Reply:
alright, i'll remove `dontBuild` and `buildInputs`.

Reply:
By removing the dependence from buildInputs, the dependence will have to be explicitly listed in the system configuration file. It is a runtime configuration.

Reply:
If it is a runtime dependency, it should be in `propagatedUserEnvPkgs` just like `gtk-engine-murrine` do, adding it to `buildInputs` shall has no observable effect (correct me if I am wrong).

Reply:
@jtojnar do you know what is the right way of dealing with runtime theming dependences in nixpkgs?

Reply:
@romildo There is not really a way outside of `propagatedUserEnvPkgs`, which is an ugly hack that will not work in some cases. Your best bet is adding a `passthru.parentThemes` or something and creating a NixOS module that deals with it. Some discussion in https://github.com/NixOS/nixpkgs/issues/73967, https://github.com/NixOS/nixpkgs/issues/39057….